### DISCUSSION TITLE
Is there Pytorch version CLIPPO?

### DISCUSSION THREAD

**Discussion by yukistavailable** [google-research/big_vision#65] [Q&A]
Thank you for releasing code for these inspiring works!
Especially, I'm interested in CLIPPO.

Are there any plans to release a Pytorch version?

**Comment by mitscha** [comment]
Thanks for your interest! We have not planned a PyTorch release for CLIPPO.

If you're interested in using the pretrained checkpoints with PyTorch, it shouldn't be too hard to convert them, as they are stored as npz files and we follow the standard ViT design. If you're interested in training in pytorch, you could just port the preprocessing function to your favorite CLIP library and adapt the code to do two forward passes through the vision encoder (one for the natural image and one for the text image).

**Comment by yukistavailable** [comment]
Thank you for the detailed explanation!

OK, I will do the conversion by myself. 😄
